middleware/varnish/detectors-varnish.tf

```diff
@@ -0,0 +1,135 @@
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.detectors_disabled)
+    notifications         = coalescelist(var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} on {{{dimensions}}}"
+  }
+}
+
+# session dropped in varnish detection
+resource "signalfx_detector" "varnish_session_dropped" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Varnish Session Dropped"
+
+  program_text = <<-EOF
+    signal = data('varnish.sess_dropped', filter=(not filter('aws_state', '{Code: 32,Name: shutting-down', '{Code: 48,Name: terminated}', '{Code: 62,Name: stopping}', '{Code: 80,Name: stopped}')) and (not filter('gcp_status', '{Code=3, Name=STOPPING}', '{Code=4, Name=TERMINATED}')) and (not filter('azure_power_state', 'PowerState/stopping', 'PowerState/stopped', 'PowerState/deallocating', 'PowerState/deallocated')) and ${module.filter-tags.filter_custom}).publish('signal')
+    detect(when(signal > ${var.varnish_session_dropped_threshold_critical})).publish('CRIT')
+EOF
+
+  rule {
+    description           = "is too high > ${var.varnish_session_dropped_threshold_critical}"
+    severity              = "Critical"
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.detectors_disabled)
+    notifications         = coalescelist(var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} on {{{dimensions}}}"
+  }
+}
+
+# session dropped in varnish detection
+resource "signalfx_detector" "varnish_cache_hit_rate" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Varnish Hit Rate"
+
+  program_text = <<-EOF
+    A = data('varnish.cache_hit', filter=(not filter('aws_state', '{Code: 32,Name: shutting-down', '{Code: 48,Name: terminated}', '{Code: 62,Name: stopping}', '{Code: 80,Name: stopped}')) and (not filter('gcp_status', '{Code=3, Name=STOPPING}', '{Code=4, Name=TERMINATED}')) and (not filter('azure_power_state', 'PowerState/stopping', 'PowerState/stopped', 'PowerState/deallocating', 'PowerState/deallocated')) and ${module.filter-tags.filter_custom}).publish('A')
+    B = data('varnish.cache_miss', filter=(not filter('aws_state', '{Code: 32,Name: shutting-down', '{Code: 48,Name: terminated}', '{Code: 62,Name: stopping}', '{Code: 80,Name: stopped}')) and (not filter('gcp_status', '{Code=3, Name=STOPPING}', '{Code=4, Name=TERMINATED}')) and (not filter('azure_power_state', 'PowerState/stopping', 'PowerState/stopped', 'PowerState/deallocating', 'PowerState/deallocated')) and ${module.filter-tags.filter_custom}).publish('B')
+    signal = (A/(A+B)).fill(0).scale(100).publish('signal')
+    detect(when(signal < ${var.varnish_cache_hit_rate_threshold_warning})).publish('WARN')
+    detect(when(signal < ${var.varnish_cache_hit_rate_threshold_critical}) and (signal > ${var.varnish_cache_hit_rate_threshold_warning})).publish('CRIT')
```

Comment:
I would use WARN instead of CRIT and MAJOR instead of Warning. hit rate is about performance issue, not availability so critical is too high.